sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -200,12 +197,9 @@ namespace Azure { namespace Core { namespace IO {
       RandomAccessFileBodyStream(int fileDescriptor, int64_t offset, int64_t length)
           : m_fileDescriptor(fileDescriptor), m_baseOffset(offset), m_length(length), m_offset(0)
       {
-        if (fileDescriptor < 1 || offset < 0 || length < 0)
-        {
-          throw std::invalid_argument(
-              "Offset and length cannot be negative. Offset: " + std::to_string(offset)
-              + ", Length: " + std::to_string(length) + ", and the file handle must be valid.");
-        }
+        AZURE_ASSERT_MSG(
+            fileDescriptor > 0 && offset >= 0 && length >= 0,
```

Comment:
Pre-existing: `0` is a perfectly valid file descriptor; this design will likely break when used with long-running background programs on POSIX that traditionally close `stdout` which makes file descriptor `0` available for reuse.